src/shadowbox/model/access_key.ts

```diff
@@ -27,6 +27,14 @@ export interface ProxyParams {
   password: string;
 }
 
+// Parameters needed to enforce an access key data quota, over a sliding window.
+export interface AccessKeyQuota {
+  // The allowed metered traffic measured in bytes.
+  quotaBytes: number;
+  // The sliding window size in hours.
+  windowHours: number;
```

Comment:
Fixing hours will limit us. I believe most admins will think in terms of days. We don't want them to enter days in hours or display days in hours. We should be able to let the admin choose the unit, and save that.

The same applies to the data field. We should be able to specify the unit. Entering bytes is error prone.

What's the proposed solution?

I guess one possibility is to allow the admin to pick the unit on input, but save the values in this normalized format. Then picking a "best unit" when displaying.

Reply:
> I guess one possibility is to allow the admin to pick the unit on input, but save the values in this normalized format. Then picking a "best unit" when displaying.

That's what I was thinking. The server manager can display any unit to the user and convert them when sending the request. It can automatically pick the best unit, or persist the user choice locally. 

These units are as fine-grained as we want to support (lowest common denominator). The alternative would be to add unit fields for both parameters, but I think it would add complexity the quota interface and API.

Reply:
Ok, I think the smart unit selection is fine for now. It's easy to extend in a way the user can express the unit.

Not necessary, but maybe we should consider a more extensible format, such as
```
AccessKeyQuota {
  quota: { bytes: number }
  window: { hours: number}
}
```
This is what we use for the [server monthly cost](https://github.com/Jigsaw-Code/outline-server/blob/363c5f16d0019a80981eee11f741c455444f91bc/src/server_manager/model/server.ts#L100)

That can later be extended in a backward-compatible way to something like:
```
AccessKeyQuota {
  quota: { bytes: number, displayUnit: GIGABYTE|MEGABYTE|...}
  window: { hours: number, displayUnit: WEEK|DAY|HOUR}
}
```

Reply:
I think your suggestion is better than hardcoding the unit in the property. Done in a233e1e.